# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/swgl/src/glsl.h
# Commit: a0b3e2669a2d
# Full Hash: a0b3e2669a2dc49c467521cb9b7966ca05e3dbc3
# Author: Ashley Hale <ahale@mozilla.com>
# Date: 2024-05-08 09:49:23
# Regressor Bug: 1824502
# File Overlap Count: 1
# Description:
#   Bug 1824502 - Implement native SVG filter rendering in WebRender r=gw
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D209646
# ==============================================================================

diff -r d1df40ba7c88 -r a0b3e2669a2d gfx/wr/swgl/src/glsl.h
--- a/gfx/wr/swgl/src/glsl.h	Wed May 08 01:43:25 2024 +0000
+++ b/gfx/wr/swgl/src/glsl.h	Wed May 08 04:38:26 2024 +0000
@@ -106,6 +106,7 @@
 
 int32_t force_scalar(I32 i) { return i[0]; }
 
+struct vec3;
 struct vec4;
 struct ivec2;
 
@@ -2287,6 +2288,12 @@
   return bvec2_scalar(notEqual(x.x, y.x), notEqual(x.y, y.y));
 }
 
+vec3 floor(vec3 v) { return vec3(floor(v.x), floor(v.y), floor(v.z)); }
+
+vec4 floor(vec4 v) {
+  return vec4(floor(v.x), floor(v.y), floor(v.z), floor(v.w));
+}
+
 struct mat4_scalar;
 
 struct mat2_scalar {